torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -1369,7 +1368,10 @@ def _linear_int8_act_int8_weight_impl(input_tensor, weight_tensor, bias):
     w_vals_int8_t = weight_tensor.tensor_impl.int_data.contiguous().t()
     w_scales = weight_tensor.tensor_impl.scale
     tmp = x_vals_int8.reshape(-1, x_vals_int8.shape[-1])
-    y_dot_scaled = int_scaled_matmul(tmp, w_vals_int8_t, x_scales.reshape(-1, 1))
+    x_scales_dtype = x_scales.dtype
+    intermediate_dtype = torch.float if x_scales_dtype == torch.half else x_scales_dtype
```

Comment:
can you add a comment on how do we decide the intermediate dtype here?

Reply:
Sure. Added.